searx/engines/il_post.py

```diff
@@ -0,0 +1,76 @@
+# SPDX-License-Identifier: AGPL-3.0-or-later
+"""Engine for Il Post, a largely independent online Italian newspaper.
+
+To use this engine add the following entry to your engines
+list in ``settings.yml``:
+
+.. code:: yaml
+
+  - name: il post
+    engine: il_post
+    shortcut: pst
+    disabled: false
+
+"""
+
+from json import loads
+from urllib.parse import urlencode
+from searx.result_types import EngineResults
+
+engine_type = 'online'
+language_support = False
+categories = ['news']
+paging = True
+page_size = 10
+
+time_range_support = True
+time_range_args = {
+    'month': 'pub_date:ultimi_30_giorni',
+    'year': 'pub_date:ultimo_anno'
+}
+
+search_api = 'https://api.ilpost.org/search/api/site_search/?'
+
+about = {
+    "website": 'https://www.ilpost.it',
+    "wikidata_id": 'Q3792882',
+    "official_api_documentation": None,
+    "use_official_api": True,
+    "require_api_key": False,
+    "results": 'JSON',
+}
+
+def request(query, params):
+    query_params = {
+        'qs': query,
+        'pg': params['pageno'],
+        'sort': "date_d",
+        'filters': 'ctype:articoli',
+    }
+
+    if params['time_range']:
+        query_params['filters'] +=  f";{time_range_args.get(params['time_range'], 'pub_date:da_sempre')}"
+
+    encoded_querystring = urlencode(query_params)
+
+
+    params['url'] = search_api + encoded_querystring
+    return params
+
+
+def response(resp) -> EngineResults:
+    res = EngineResults()
+    json_data = loads(resp.text)
+
+    for result in json_data['docs']:
+
+        res.append(
+            {
+                'url': result['link'],
+                'title': result['title'],
+                'content': result['summary'] or None,
+                'thumbnail': result['image'] or None,
```

Comment:
Is the `or None` for the case that it's an empty string? Otherwise, if `result['image']` is None, this expression is `None or None` and thus redundant.

Reply:
I do remember a couple of cases where the content and/or the thumbnail was absent, but I couldn't find those results again to test, I don't know if the field was empty.
I thought about using an empty string for both fields as you proposed, but I couldn't decide and ended up doing like this. Is there a recommended/idiomatic way of doing this in the project's conventions? I also thought Python's "or" is short-circuited so the possible redundancy wouldn't be a problem.

Please let me know what you think about this and thank you for your comments and tips, my Python is a little bit rusty :)

Reply:
For SearXNGs `content` field: This must be set in order for the result to be displayed. If it's none, the result will be hidden, thus the empty string here. (That's just because of some SearXNG internal stuff, I remember we had plans to change that behavior in the future)

Reply:
For the thumbnail, we should probably just use `result.get('image')`, which defaults to None if the field is unset.

If using `result['image']` although the result doesn't contain an image field, this call would raise an exception and crash the engine.